### PR TITLE
After sending a TCP packet, swap the source and target IP address ( I…

### PR DESCRIPTION
**Pull request by htibosch**
Description
-----------
When TCP sends out a packet in `prvTCPReturnPacket()`, it temporarily swaps the IPv6 addresses source and target. When the packet is part of the socket space, it will be reused later on. In that case, it is important to restore the value of `pxIPHeader_IPv6->xSourceAddress`

Related Issue
-----------
In some cases, a TCP socket would send a packet to its own IPv6-address.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AniruddhaKanhere — DISMISSED** [review]

**Review by cobusve — DISMISSED** [review]

File: FreeRTOS_TCP_IP.c [hunk 1]
```diff
@@ -1153,9 +1150,12 @@
                 #if ( ipconfigUSE_IPv6 != 0 )
                     if( ( pxSocket != NULL ) && ( pxSocket->bits.bIsIPv6 != pdFALSE_UNSIGNED ) )
```
Comment (gkwicker) [inline FreeRTOS_TCP_IP.c:1151]: The v4 logic on line 1158 verifies that `pxIPHeader != NULL` prior to dereferencing it at line 1160.  Should we do the same at line 1151, e.g.:

```c
if( ( pxSocket != NULL ) && 
    ( pxSocket->bits.bIsIPv6 != pdFALSE_UNSIGNED ) && 
    ( pxIPHeader_IPv6 != NULL ) )
```

**Review by gkwicker — COMMENTED** [review]
Just one question, see my comment.

File: FreeRTOS_TCP_IP.c (hunk 1 above)
Comment (htibosch) [reply FreeRTOS_TCP_IP.c:1151]: Thanks Gary, I added the extra test for `pxIPHeader_IPv6` not being NULL.

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by cobusve — APPROVED** [review]
